js/controller.js

```diff
@@ -993,8 +1000,40 @@ OO.plugin("Html5Skin", function (OO, _, $, W) {
         clearTimeout(this.state.timer);
         this.state.timer = null;
       }
+    },
+
+    enableAspectRatio: function() {
+      //auto set aspect ratio
+      if(this.state.isInitialPlay && (!this.skin.props.skinConfig.responsive.aspectRatio || this.skin.props.skinConfig.responsive.aspectRatio == "auto")) {
```

Comment:
I would recommend that aspectRatio setting in skin.json is changed into autoAspectRatio with a type of Boolean.